gpflow/kernels/misc.py

```diff
@@ -98,51 +98,73 @@ def K_diag(self, X):
 
 
 class Coregion(Kernel):
-    def __init__(self, output_dim, rank, active_dims=None):
-        """
-        A Coregionalization kernel. The inputs to this kernel are _integers_
-        (we cast them from floats as needed) which usually specify the
-        *outputs* of a Coregionalization model.
+    """
+    A Coregionalization kernel. The inputs to this kernel are _integers_ (we
+    cast them from floats as needed) which usually specify the *outputs* of a
+    Coregionalization model.
+
+    The kernel function is an indexing of a positive-definite matrix:
 
-        The parameters of this kernel, W, kappa, specify a positive-definite
-        matrix B.
+      K(x, y) = B[x, y] .
 
-          B = W W^T + diag(kappa) .
+    To ensure that B is positive-definite, it is specified by the two
+    parameters of this kernel, W and kappa:
 
-        The kernel function is then an indexing of this matrix, so
+      B = W Wᵀ + diag(kappa) .
 
-          K(x, y) = B[x, y] .
+    We refer to the size of B as "output_dim x output_dim", since this is the
+    number of outputs in a coregionalization model. We refer to the number of
+    columns on W as 'rank': it is the number of degrees of correlation between
+    the outputs.
 
-        We refer to the size of B as "num_outputs x num_outputs", since this is
-        the number of outputs in a coregionalization model. We refer to the
-        number of columns on W as 'rank': it is the number of degrees of
-        correlation between the outputs.
+    NB. There is a symmetry between the elements of W, which creates a local
+    minimum at W=0. To avoid this, it is recommended to initialize the
+    optimization (or MCMC chain) using a random W.
+    """
 
-        NB. There is a symmetry between the elements of W, which creates a
-        local minimum at W=0. To avoid this, it's recommended to initialize the
-        optimization (or MCMC chain) using a random W.
+    def __init__(self, output_dim: int, rank: int, **kwargs):
+        """
+        :param output_dim: number of outputs expected (0 <= X < output_dim)
+        :param rank: number of degrees of correlation between outputs
         """
 
         # assert input_dim == 1, "Coregion kernel in 1D only"
-        super().__init__(active_dims)
+        super().__init__(**kwargs)
 
         self.output_dim = output_dim
         self.rank = rank
-        W = np.zeros((self.output_dim, self.rank))
+        W = 0.1 * np.ones((self.output_dim, self.rank))
         kappa = np.ones(self.output_dim)
         self.W = Parameter(W)
         self.kappa = Parameter(kappa, transform=positive())
 
+    def compute_B(self):
```

Comment:
can this be private?

Reply:
it's actually useful for users of this kernel! e.g. see the coregion notebook.

maybe we want to call it something more descriptive ... `output_covariances()` and `output_variances()` maybe?

Reply:
i thought the kernel was the covariance. what's the covariance then? and the variance?

Reply:
> maybe we want to call it something more descriptive ... output_covariances() and output_variances() maybe?

Yes, the name can be improved. copute_B looks cryptic :). I'd propose singular names `output_covariance` & `output_variance`.

Reply:
or
```python3
@property
def covariance(self) -> tf.Tensor:
   ...
```
etc

Reply:
as this does computation, I think it's important to not hide that behind a property. renamed to output_covariance as I suggested.

Reply:
Everything does computation. I've been thinking about this in general for a few days now, as a more general question, and I've come to the conclusion that naming shouldn't be related to performance, because performance is typically an implementation detail and relied heavily on architecture.

As an aside, I can't as a reader tell if `output_` is a noun or verb